server/service/team_schedule.go

```diff
@@ -39,6 +39,9 @@ func (svc Service) GetTeamScheduledQueries(ctx context.Context, teamID uint, opt
 	if err := svc.authz.Authorize(ctx, &fleet.Pack{}, fleet.ActionRead); err != nil {
 		return nil, err
 	}
+	if err := svc.authz.TeamAuthorize(ctx, teamID, fleet.ActionRead); err != nil {
```

Comment:
When I realized we were missing these checks, I also added them to the team schedule calls.